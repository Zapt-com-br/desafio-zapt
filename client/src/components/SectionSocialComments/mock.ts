export default [
  {
    user: {
      name: 'Alan Gabriel',
      avatar_url: '/img/profile.jpg',
      username: 'eualangabriel',
      social_provider: 'Twitter'
    },
    testimonial: {
      text:
        'Eu disse que seria a melhor pessoa para ajudar a criar uma ambiente foda, para construir coisas incríveis'
    }
  },
  {
    user: {
      name: 'Alan Gabriel',
      avatar_url: '/img/profile.jpg',
      username: 'eualangabriel',
      social_provider: 'Twitter'
    },
    testimonial: {
      text:
        'Depois que entrei pra Zapt consegui ter clareza sobre o futuro da minha carreira'
    }
  },
  {
    user: {
      name: 'Alan Gabriel',
      avatar_url: '/img/profile.jpg',
      username: 'eualangabriel',
      social_provider: 'Twitter'
    },
    testimonial: {
      text:
        'A oportunidade de ter um mentor que é especialista frontend é um passo gigante pra acelerar a carreira'
    }
  },
  {
    user: {
      name: 'Alan Gabriel',
      avatar_url: '/img/profile.jpg',
      username: 'eualangabriel',
      social_provider: 'Twitter'
    },
    testimonial: {
      text:
        'Ceis também querem um caneca da Zapt? Tem que ser ousado pra conseguir em! hahaha'
    }
  },
  {
    user: {
      name: 'Alan Gabriel',
      avatar_url: '/img/profile.jpg',
      username: 'eualangabriel',
      social_provider: 'Twitter'
    },
    testimonial: {
      text:
        'A oportunidade de ter um mentor que é especialista frontend é um passo gigante pra acelerar a carreira'
    }
  },
  {
    user: {
      name: 'Alan Gabriel',
      avatar_url: '/img/profile.jpg',
      username: 'eualangabriel',
      social_provider: 'Twitter'
    },
    testimonial: {
      text:
        'Depois que entrei pra Zapt consegui ter clareza sobre o futuro da minha carreira'
    }
  },
  {
    user: {
      name: 'Alan Gabriel',
      avatar_url: '/img/profile.jpg',
      username: 'eualangabriel',
      social_provider: 'Twitter'
    },
    testimonial: {
      text:
        'A oportunidade de ter um mentor que é especialista frontend é um passo gigante pra acelerar a carreira'
    }
  },
  {
    user: {
      name: 'Alan Gabriel',
      avatar_url: '/img/profile.jpg',
      username: 'eualangabriel',
      social_provider: 'Twitter'
    },
    testimonial: {
      text:
        'A oportunidade de ter um mentor que é especialista frontend é um passo gigante pra acelerar a carreira'
    }
  },
  {
    user: {
      name: 'Alan Gabriel',
      avatar_url: '/img/profile.jpg',
      username: 'eualangabriel',
      social_provider: 'Twitter'
    },
    testimonial: {
      text:
        'Ceis também querem um caneca da Zapt? Tem que ser ousado pra conseguir em! hahaha'
    }
  }
]
