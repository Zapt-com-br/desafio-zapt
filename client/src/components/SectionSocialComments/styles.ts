import styled, { css } from 'styled-components'

export const Wrapper = styled.div`
  ${({ theme }) => css`
    width: 100%;
    max-width: ${theme.grid.container};
    margin: 0 auto;
    padding: 0 ${theme.spacings.small};

    h2 {
      text-align: center;
      margin-bottom: ${theme.spacings.xlarge};
    }

    @media (max-width: 550px) {
      h2 {
        margin-bottom: ${theme.spacings.medium};
      }
    }
  `}
`

export const WrapperCards = styled.div`
  ${({ theme }) => css`
    column-count: 3;

    > div {
      margin-bottom: ${theme.spacings.small};
    }

    @media (max-width: 768px) {
      column-count: 2;
    }

    @media (max-width: 450px) {
      column-count: 1;
    }
  `}
`
