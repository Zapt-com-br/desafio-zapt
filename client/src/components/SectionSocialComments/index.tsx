import CardSocialComment from 'components/CardSocialComment'
import Heading from 'components/Heading'
import mock from './mock'
import * as S from './styles'

const SectionSocialComments = () => (
  <S.Wrapper>
    <Heading level={2} size="huge">
      O que est√£o falando nas redes sociais
    </Heading>
    <S.WrapperCards data-testid="wraper-social-cards">
      {mock.map((item, index) => (
        <CardSocialComment key={`${item.user.username}-${index}`} {...item} />
      ))}
    </S.WrapperCards>
  </S.Wrapper>
)

export default SectionSocialComments
