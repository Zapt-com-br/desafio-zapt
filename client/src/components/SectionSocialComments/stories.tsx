import { Story, Meta } from '@storybook/react'
import SectionSocialComments from '.'

export default {
  title: 'SectionSocialComments',
  component: SectionSocialComments
} as Meta

export const Default: Story = () => <SectionSocialComments />
