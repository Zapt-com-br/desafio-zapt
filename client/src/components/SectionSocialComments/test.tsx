import { screen } from '@testing-library/react'
import renderWithTheme from 'utils/tests/test-util'

import SectionSocialComments from '.'

describe('<SectionSocialComments />', () => {
  it('should render the button correctly', () => {
    const { container } = renderWithTheme(<SectionSocialComments />)

    const heading = screen.getByRole('heading', {
      name: /O que est√£o falando nas redes sociais/i
    })

    expect(heading).toBeInTheDocument()

    expect(screen.getByTestId('wraper-social-cards')).toHaveStyle({
      'column-count': 3
    })

    expect(screen.getByTestId('wraper-social-cards')).toHaveStyleRule(
      'column-count',
      '2',
      {
        media: '(max-width: 768px)'
      }
    )

    expect(screen.getByTestId('wraper-social-cards')).toHaveStyleRule(
      'column-count',
      '1',
      {
        media: '(max-width: 450px)'
      }
    )

    expect(container.firstChild).toMatchSnapshot()
  })
})
