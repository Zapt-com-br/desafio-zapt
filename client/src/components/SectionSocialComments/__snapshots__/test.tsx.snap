// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SectionSocialComments /> should render the button correctly 1`] = `
.c3 {
  width: 100%;
  max-width: 38.4rem;
  -webkit-break-inside: avoid-column;
  break-inside: avoid-column;
  padding: 2.4rem 1.6rem;
  border-radius: 0.4rem;
  background: #EBF4FD;
}

.c3 p {
  font-size: 1.6rem;
  line-height: 137%;
  -webkit-letter-spacing: -0.01em;
  -moz-letter-spacing: -0.01em;
  -ms-letter-spacing: -0.01em;
  letter-spacing: -0.01em;
  color: #011FBB;
}

.c3 p::after {
  content: '”';
}

.c3 p::before {
  content: '“';
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 12px;
  margin-top: 1.6rem;
}

.c4 img {
  width: 36px;
  height: 36px;
  object-fit: cover;
  border-radius: 50%;
  border: 1px solid #334BC8;
}

.c4 span {
  font-weight: 500;
  font-size: 1.4rem;
  line-height: 129%;
  -webkit-letter-spacing: -0.01em;
  -moz-letter-spacing: -0.01em;
  -ms-letter-spacing: -0.01em;
  letter-spacing: -0.01em;
  color: #001C33;
}

.c1 {
  font-size: 4.0rem;
  color: #000000;
  font-weight: 500;
}

.c0 {
  width: 100%;
  max-width: 124.8rem;
  margin: 0 auto;
  padding: 0 2.4rem;
}

.c0 h2 {
  text-align: center;
  margin-bottom: 4.8rem;
}

.c2 {
  -webkit-column-count: 3;
  column-count: 3;
}

.c2 > div {
  margin-bottom: 2.4rem;
}

@media (max-width:550px) {
  .c1 {
    font-size: calc(4.0rem / 1.3);
    line-height: 1.3;
  }
}

@media (max-width:550px) {
  .c0 h2 {
    margin-bottom: 3.2rem;
  }
}

@media (max-width:768px) {
  .c2 {
    -webkit-column-count: 2;
    column-count: 2;
  }
}

@media (max-width:450px) {
  .c2 {
    -webkit-column-count: 1;
    column-count: 1;
  }
}

<div
  class="c0"
>
  <h2
    class="c1"
    color="#000000"
    font-weight="500"
  >
    O que estão falando nas redes sociais
  </h2>
  <div
    class="c2"
    data-testid="wraper-social-cards"
  >
    <div
      class="c3"
    >
      <p>
        Eu disse que seria a melhor pessoa para ajudar a criar uma ambiente foda, para construir coisas incríveis
      </p>
      <div
        class="c4"
        title="Alan Gabriel via Twitter"
      >
        <div
          style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
        >
          <div
            style="box-sizing: border-box; display: block; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              role="presentation"
              src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMzYiIGhlaWdodD0iMzYiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
              style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
            />
          </div>
          <noscript />
          <img
            alt="Foto de perfil de Alan Gabriel"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
          />
        </div>
        <span>
          @
          eualangabriel
        </span>
      </div>
    </div>
    <div
      class="c3"
    >
      <p>
        Depois que entrei pra Zapt consegui ter clareza sobre o futuro da minha carreira
      </p>
      <div
        class="c4"
        title="Alan Gabriel via Twitter"
      >
        <div
          style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
        >
          <div
            style="box-sizing: border-box; display: block; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              role="presentation"
              src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMzYiIGhlaWdodD0iMzYiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
              style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
            />
          </div>
          <noscript />
          <img
            alt="Foto de perfil de Alan Gabriel"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
          />
        </div>
        <span>
          @
          eualangabriel
        </span>
      </div>
    </div>
    <div
      class="c3"
    >
      <p>
        A oportunidade de ter um mentor que é especialista frontend é um passo gigante pra acelerar a carreira
      </p>
      <div
        class="c4"
        title="Alan Gabriel via Twitter"
      >
        <div
          style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
        >
          <div
            style="box-sizing: border-box; display: block; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              role="presentation"
              src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMzYiIGhlaWdodD0iMzYiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
              style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
            />
          </div>
          <noscript />
          <img
            alt="Foto de perfil de Alan Gabriel"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
          />
        </div>
        <span>
          @
          eualangabriel
        </span>
      </div>
    </div>
    <div
      class="c3"
    >
      <p>
        Ceis também querem um caneca da Zapt? Tem que ser ousado pra conseguir em! hahaha
      </p>
      <div
        class="c4"
        title="Alan Gabriel via Twitter"
      >
        <div
          style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
        >
          <div
            style="box-sizing: border-box; display: block; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              role="presentation"
              src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMzYiIGhlaWdodD0iMzYiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
              style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
            />
          </div>
          <noscript />
          <img
            alt="Foto de perfil de Alan Gabriel"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
          />
        </div>
        <span>
          @
          eualangabriel
        </span>
      </div>
    </div>
    <div
      class="c3"
    >
      <p>
        A oportunidade de ter um mentor que é especialista frontend é um passo gigante pra acelerar a carreira
      </p>
      <div
        class="c4"
        title="Alan Gabriel via Twitter"
      >
        <div
          style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
        >
          <div
            style="box-sizing: border-box; display: block; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              role="presentation"
              src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMzYiIGhlaWdodD0iMzYiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
              style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
            />
          </div>
          <noscript />
          <img
            alt="Foto de perfil de Alan Gabriel"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
          />
        </div>
        <span>
          @
          eualangabriel
        </span>
      </div>
    </div>
    <div
      class="c3"
    >
      <p>
        Depois que entrei pra Zapt consegui ter clareza sobre o futuro da minha carreira
      </p>
      <div
        class="c4"
        title="Alan Gabriel via Twitter"
      >
        <div
          style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
        >
          <div
            style="box-sizing: border-box; display: block; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              role="presentation"
              src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMzYiIGhlaWdodD0iMzYiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
              style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
            />
          </div>
          <noscript />
          <img
            alt="Foto de perfil de Alan Gabriel"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
          />
        </div>
        <span>
          @
          eualangabriel
        </span>
      </div>
    </div>
    <div
      class="c3"
    >
      <p>
        A oportunidade de ter um mentor que é especialista frontend é um passo gigante pra acelerar a carreira
      </p>
      <div
        class="c4"
        title="Alan Gabriel via Twitter"
      >
        <div
          style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
        >
          <div
            style="box-sizing: border-box; display: block; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              role="presentation"
              src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMzYiIGhlaWdodD0iMzYiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
              style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
            />
          </div>
          <noscript />
          <img
            alt="Foto de perfil de Alan Gabriel"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
          />
        </div>
        <span>
          @
          eualangabriel
        </span>
      </div>
    </div>
    <div
      class="c3"
    >
      <p>
        A oportunidade de ter um mentor que é especialista frontend é um passo gigante pra acelerar a carreira
      </p>
      <div
        class="c4"
        title="Alan Gabriel via Twitter"
      >
        <div
          style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
        >
          <div
            style="box-sizing: border-box; display: block; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              role="presentation"
              src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMzYiIGhlaWdodD0iMzYiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
              style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
            />
          </div>
          <noscript />
          <img
            alt="Foto de perfil de Alan Gabriel"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
          />
        </div>
        <span>
          @
          eualangabriel
        </span>
      </div>
    </div>
    <div
      class="c3"
    >
      <p>
        Ceis também querem um caneca da Zapt? Tem que ser ousado pra conseguir em! hahaha
      </p>
      <div
        class="c4"
        title="Alan Gabriel via Twitter"
      >
        <div
          style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
        >
          <div
            style="box-sizing: border-box; display: block; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              role="presentation"
              src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMzYiIGhlaWdodD0iMzYiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
              style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
            />
          </div>
          <noscript />
          <img
            alt="Foto de perfil de Alan Gabriel"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
          />
        </div>
        <span>
          @
          eualangabriel
        </span>
      </div>
    </div>
  </div>
</div>
`
